File: Py/Classi/Scuola/docente.py
from persona import Persona
from corso import Corso

class Docente(Persona):

    # Costruttore
    def __init__(self, nome, cognome, matricola):
        super().__init__(nome, cognome, matricola)
        self.corsi = []

    # Utility
    def addCorso(self, nomeCorso, matricola, durata):
        corso = Corso(nomeCorso, matricola, durata)
        self.corsi.append(corso)

    # Getters
    def getCorsi(self):
        return self.corsi
    
    def getCorsi(self):
        for corso in self.corsi:
            print(f'Nome: {corso.nomeCorso}, Marticola: {corso.matricola}, Durata: {corso.durata}')

File: Java/Scuola/src/Docente.java
import java.util.ArrayList;
import java.util.List;

public class Docente {

    private List<Object> corsi = new ArrayList<Object>();

    // Constructor
    public Docente(List<Object> corsiList) {
        this.corsi = corsiList;
    }

    // Getters
    public void getCorsi() {
        for (Object corso : corsi) {
            // System.out.println( corso.nomeCorso + " " + corso.matricola + " " + corso.durataAnni);
        }
    }

    // Setters
    public void setCorsi(List<Object> corsiList) {
        this.corsi = corsiList;
    }

    // Utility
    public void addCorso(Object corsoObject) {
        this.corsi.add(corsoObject);
    }

}
